controllers/services/auth/auth_controller_actions.go

```diff
@@ -0,0 +1,231 @@
+			Path: AdminGroupRoleTemplate,
+		},
+		{
+			FS : resourcesFS,
+			Path: AllowedGroupRoleTemplate,
+		},
+	}
+
+	return nil
+}
+
+func copyGroups(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ai, ok := rr.Instance.(*serviceApi.Auth)
+	if !ok {
+		return errors.New("instance is not of type *services.Auth")
+	}
+
+	newGroups := false
+
+	// Get groups from the dashboardConfig
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+
+	if err := rr.Client.Get(ctx, client.ObjectKey{
+		Name: "odh-dashboard-config",
+		Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+	}, odhObject); err == nil {
+		// We found the dashboardConfig
+		// If there is an error here we are ignoring it because we don't need the 
+		// dashboardConfig to exist
+		foundGroups, _, err := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
+		if err != nil {
+			return nil
+		}
+
+		adming := foundGroups["adminGroups"]
+		if len(adming) != 0 {
+			for _, e := range strings.Split(adming, ",") {
+				e = strings.TrimSpace(e)
+				// If the group is already in the list then ignore it
+				if !slices.Contains(ai.Spec.AdminGroups, e) {
+					ai.Spec.AdminGroups = append(ai.Spec.AdminGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+
+		allowedg := foundGroups["allowedGroups"]
+		if len(allowedg) != 0 {
+			for _, e := range strings.Split(allowedg, ",") {
+				// If the group is already in the list then ignore it
+				if !slices.Contains(ai.Spec.AllowedGroups, e) {
+					ai.Spec.AllowedGroups = append(ai.Spec.AllowedGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+	}
+		
+	if len(ai.Spec.AdminGroups) == 0 {
+		// adminGroups is empty so add the default
+		ai.Spec.AdminGroups = append(ai.Spec.AdminGroups, dashboard.GetAdminGroup())
+		newGroups = true
+	}
+
+	if len(ai.Spec.AllowedGroups) == 0 {
+		// allowedGroups is empty so add the default
+		ai.Spec.AllowedGroups = append(ai.Spec.AllowedGroups, "system:authenticated")
+		newGroups = true
+	}
+
+	// only update if we found a new group in the list
+	if newGroups {
+		err := rr.Client.Update(ctx, ai)
+		if err != nil {
+			return errors.New("error adding groups to Auth CR")
+		}
+	}
+
+	return nil
+}
+
+func bindRole(ctx context.Context, rr *odhtypes.ReconciliationRequest, groups []string, roleBindingName string, roleName string) error {
+	var groupsToBind []rbacv1.Subject
+	for _, e := range groups {
+		rs := rbacv1.Subject{
+			Kind:     "Group",
+			APIGroup: "rbac.authorization.k8s.io",
+			Name:     e,
+		}
+		groupsToBind = append(groupsToBind, rs)
+	}
+
+	rb := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+		},
+		Subjects: groupsToBind,
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     roleName,
+		},
+	}
+	err := rr.Client.Update(ctx, rb)
```

Comment:
wonder if this can be delegated to `action/deploy` or if you found any issue with that

Reply:
Yeah I just couldn't find an easy way to do it. Calling this line seemed more straightforward, presumably i'm just misunderstanding something though?

Reply:
if you add the `RoleBinding` to the `Resources`, then the `deploy` action (you probably need to add it to the controller) should take care to deploy it. However, I don't know if that action is doing more since it was originally designed for components, so maybe we can give it a try and see what should be amended.

Reply:
I was missing some permissions and had to move some ordering around but it's using that deploy action now.